src/components/side_panel/chart/bar_chart/bar_chart_config_panel.xml

```diff
@@ -1,66 +1,36 @@
 <templates>
   <t t-name="o-spreadsheet-BarConfigPanel">
     <div>
-      <div class="o-section pt-0">
-        <label class="o-checkbox">
-          <input
-            type="checkbox"
-            name="stacked"
-            t-att-checked="props.definition.stacked"
-            t-on-change="onUpdateStacked"
-            class="align-middle"
-          />
-          Stacked barchart
-        </label>
-      </div>
-      <div class="o-section o-data-series">
-        <div class="o-section-title">Data Series</div>
-        <SelectionInput
-          ranges="() => this.getDataSeriesRanges()"
-          required="true"
-          onSelectionChanged="(ranges) => this.onDataSeriesRangesChanged(ranges)"
-          onSelectionConfirmed="() => this.onDataSeriesConfirmed()"
+      <Section class="'pt-0'">
+        <Checkbox
+          name="'stacked'"
+          label="stackedLabel"
+          value="props.definition.stacked"
+          onChange.bind="onUpdateStacked"
         />
-      </div>
-      <div class="o-section o-data-labels">
-        <div class="o-section-title">Categories / Labels</div>
-        <SelectionInput
-          ranges="() => [this.getLabelRange()]"
-          isInvalid="isLabelInvalid"
-          hasSingleRange="true"
-          onSelectionChanged="(ranges) => this.onLabelRangeChanged(ranges)"
-          onSelectionConfirmed="() => this.onLabelRangeConfirmed()"
+      </Section>
+      <ChartDataSeries
+        ranges="() => this.getDataSeriesRanges()"
+        onSelectionChanged="(ranges) => this.onDataSeriesRangesChanged(ranges)"
+        onSelectionConfirmed="() => this.onDataSeriesConfirmed()"
+      />
+      <ChartLabelRange
+        range="() => this.getLabelRange()"
+        isInvalid="isLabelInvalid"
```

Comment:
No idea 🤷‍♂️  It's a good question, because the use cases I checked, `isInvalid` is a simple check of the validity of the range, that could be done in the selection input component.
I created a task (3647835) to investigate more